compaction/task_manager_module.hh

```diff
@@ -31,6 +31,27 @@ protected:
     virtual future<> run() override = 0;
 };
 
+class major_compaction_task_impl : public compaction_task_impl {
+public:
+    major_compaction_task_impl(tasks::task_manager::module_ptr module,
+            tasks::task_id id,
+            unsigned sequence_number,
+            std::string keyspace,
+            std::string table,
+            std::string entity,
+            tasks::task_id parent_id) noexcept
+        : compaction_task_impl(module, id, sequence_number, std::move(keyspace), std::move(table), std::move(entity), parent_id)
+    {
+        // FIXME: add progress units
+    }
+
+    virtual std::string type() const override {
+        return "major compaction";
+    }
```

Comment:
How is this going to be used? Would it make sense to have an enum here instead of a string?

Reply:
It is used in task manager api, from which user gets stringified status (containing type). We could have an enum type attribute, which is serialized in type() method. I am not sure it's worth it.

Reply:
Ok, it might not worth converting to an enum then.